src/Conversion/ONNXToKrnl/Sequence/SequenceInsert.cpp

```diff
@@ -0,0 +1,138 @@
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op->getLoc();
+    ONNXSequenceInsertOpAdaptor operandAdaptor(operands);
+    ONNXSequenceInsertOp thisOp = dyn_cast<ONNXSequenceInsertOp>(op);
+    MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
+    IndexExprScope IEScope(&rewriter, loc);
+
+    // ONNXSequenceInsertOp sequenceInsertOp = cast<ONNXSequenceInsertOp>(op);
+
+    // Get element type for seq from the output
+
+    auto input_sequence = operandAdaptor.input_sequence();
+    auto dimSize = rewriter.create<memref::DimOp>(
+        loc, input_sequence, create.math.constantIndex(0));
+    SymbolIndexExpr boundIE(dimSize);
+
+    // Can not use this type because the output element type may be more precise
+    // This issue may be resolved in shape inference by backwards propagation
+    // In some sense, the type info for seq is not consistent at SequenceInsert
+    // The code of type converter for Seq is duplicated
+    // auto seqElementType =
+    // input_sequence.getType().cast<MemRefType>().getElementType();
+
+    ShapedType seqElementType =
+        thisOp.getResult().getType().cast<SeqType>().getElementType();
+    Type elementType = seqElementType.getElementType();
+    Type seqElementConvertedType;
+    if (seqElementType.hasRank()) {
+      seqElementConvertedType =
+          MemRefType::get(seqElementType.getShape(), elementType);
+    } else {
+      seqElementConvertedType = UnrankedMemRefType::get(elementType, 0);
+    }
+
+    SmallVector<int64_t, 1> dims;
+    // Number of element in seq may be statically known from shape inference
+    dims.emplace_back(thisOp.getResult().getType().cast<SeqType>().getLength());
+    llvm::ArrayRef<int64_t> shape(dims.data(), dims.size());
+    MemRefType outputMemRefType =
+        MemRefType::get(shape, seqElementConvertedType);
+
+    // Output sequence has one more element
+    auto outputBound = boundIE + 1;
+    SmallVector<IndexExpr, 1> ubsIE;
+    ubsIE.emplace_back(outputBound);
+    Value alloc =
+        insertAllocAndDeallocSimple(rewriter, op, outputMemRefType, loc, ubsIE);
+
+    // Fill the output sequence
+
+    IndexExpr positionIE;
+    if (isFromNone(operandAdaptor.position())) {
+      // Insert at the end of the sequence
+      // Could be optimized as: Copy the input sequence and attach input tensor
+      // at the end But the size for KrnlMemcpy is integer, not Value
+      // memref::copy requires that source and destination have the same shape
+      positionIE = boundIE;
+    } else {
+      positionIE = SymbolIndexExpr(create.krnl.load(operandAdaptor.position()));
+      // Handle the negative position
+      positionIE =
+          IndexExpr::select(positionIE < 0, positionIE + boundIE, positionIE);
+    }
+
+    // Copy elements before the insertion position
+    KrnlBuilder createKrnl(rewriter, loc);
```

Comment:
This function have already a KrnlBuilder (`create.krnl`) that can be used instead of this one.